tests/e2e/test_api.py

```diff
@@ -50,3 +50,64 @@ def test_invalid_question() -> None:
             "know how to handle."
         }
     }
+
+
+def test_valid_question_noyaml() -> None:
+    """Check the REST API /v1/query with POST HTTP method for valid question and no yaml."""
+    response = client.post(
+        "/v1/query",
+        json={"conversation_id": "1234", "query": "what is kubernetes?"},
+        timeout=90,
+    )
+    print(vars(response))
+    assert response.status_code == requests.codes.ok
+    json_response = response.json()
+    json_response["query"] == "what is kubernetes?"
+    json_response["conversation_id"] == "1234"
+    assert "Summary:" in json_response["response"]
+    assert "kubernetes" in json_response["response"].lower()
+
+
+def test_valid_question_yaml() -> None:
+    """Check the REST API /v1/query with POST HTTP method for valid question and yaml response."""
+    response = client.post(
+        "/v1/query",
+        json={
+            "conversation_id": "1234",
+            "query": "write a deployment yaml for the mongodb image",
+        },
+        timeout=90,
+    )
+    print(vars(response))
+    assert response.status_code == requests.codes.ok
+    json_response = response.json()
+    json_response["query"] == "write a deployment yaml for the mongodb image"
+    json_response["conversation_id"] == "1234"
+
+    assert "apiVersion: apps/v1" in json_response["response"]
+    assert "kind: Deployment" in json_response["response"]
+    assert "image: mongo " in json_response["response"]
+
+
+def test_valid_question_conversation_id() -> None:
+    """Check the REST API /v1/query with POST HTTP method for same conversation id."""
+    response = client.post(
+        "/v1/query", json={"query": "what is kubernetes?"}, timeout=90
+    )
+    print(vars(response))
+    assert response.status_code == requests.codes.ok
+    json_response = response.json()
+    conversion_id = json_response["conversation_id"]
+
+    # sending second request with same conversation id
```

Comment:
i don't see what this accomplishes/tests.  Nothing in this test would fail if you sent the second request without the previous conversation id.

Reply:
Also it is probably premature to be trying to test the conversation history/reuse functionality of OLS.

Reply:
@bparees : Should we revisit the conversation_id test after the story OLS-110 is complete and mark this story as done?

Reply:
yeah that might be best, there's some cache reuse implemented in OLS today, but probably not enough to be able to write a good test against.

but please delete this test from your PR, i don't want someone thinking we have coverage for the cache based on this test which doesn't really check anything